controllers/alphacontrollers/alert_controller.go

```diff
@@ -65,188 +65,171 @@ type AlertReconciler struct {
+			return resultError, err
+		}
+		return resultOk, nil
+	}
 
-			log.V(1).Info("Alert was deleted", "Alert ID", alertId)
-			// remove our finalizer from the list and update it.
-			controllerutil.RemoveFinalizer(alertCRD, alertFinalizerName)
-			if err := r.Update(ctx, alertCRD); err != nil {
-				log.Error(err, "Error on updating alert", "Name", alertCRD.Name, "Namespace", alertCRD.Namespace)
-				return ctrl.Result{}, err
-			}
+	if !alert.ObjectMeta.DeletionTimestamp.IsZero() {
+		err = r.delete(ctx, log, alert)
+		if err != nil {
+			log.Error(err, "Error on deleting alert")
+			return resultError, err
 		}
+		return resultOk, nil
+	}
 
-		// Stop reconciliation as the item is being deleted
-		return ctrl.Result{}, nil
+	err = r.update(ctx, log, alert)
+	if err != nil {
+		log.Error(err, "Error on updating alert")
+		return resultError, err
 	}
 
-	var (
-		notFound    bool
-		err         error
-		actualState *coralogixv1alpha1.AlertStatus
-	)
-	if id := alertCRD.Status.ID; id == nil {
-		log.V(1).Info("alert wasn't created")
-		notFound = true
-	} else {
-		getAlertResp, err := alertsClient.GetAlert(ctx, &alerts.GetAlertByUniqueIdRequest{Id: wrapperspb.String(*id)})
-		switch {
-		case status.Code(err) == codes.NotFound:
-			log.V(1).Info("alert doesn't exist in Coralogix backend", "ID", id)
-			notFound = true
-		case err != nil:
-			log.Error(err, "Received an error while getting Alert")
-			return ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}, err
-		case err == nil:
-			actualState, err = flattenAlert(ctx, getAlertResp.GetAlert(), alertCRD.Spec)
-			if err != nil {
-				log.Error(err, "Received an error while flattened Alert")
-				return ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}, err
-			}
-		}
+	return resultOk, nil
+}
+
+func (r *AlertReconciler) update(ctx context.Context,
+	log logr.Logger,
+	alert *coralogixv1alpha1.Alert) error {
+	remoteAlert, err := r.CoralogixClientSet.Alerts().GetAlert(ctx, &alerts.GetAlertByUniqueIdRequest{
+		Id: wrapperspb.String(*alert.Status.ID),
+	})
+
+	if err != nil {
+		log.Error(err, "Error on getting alert")
+		return err
 	}
 
-	if notFound {
-		if alertCRD.Spec.Labels == nil {
-			alertCRD.Spec.Labels = make(map[string]string)
-		}
-		alertCRD.Spec.Labels["managed-by"] = "coralogix-operator"
-		if err := r.Client.Update(ctx, alertCRD); err != nil {
-			log.Error(err, "Error on updating alert", "Name", alertCRD.Name, "Namespace", alertCRD.Namespace)
-			return ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}, err
-		}
+	status, err := getStatus(ctx, remoteAlert.GetAlert(), alert.Spec)
+	if err != nil {
+		log.Error(err, "Error on flattening alert")
+		return err
+	}
 
-		createAlertReq, err := alertCRD.Spec.ExtractCreateAlertRequest(ctx)
-		if err != nil {
-			log.Error(err, "Bad request for creating alert", "Name", alertCRD.Name, "Namespace", alertCRD.Namespace)
-			return ctrl.Result{}, err
-		}
+	equal, _ := alert.Spec.DeepEqual(&status)
+	if equal {
+		return nil
+	}
```

Comment:
Micronit: Could be simplified

```suggestion
	if equal, _ := alert.Spec.DeepEqual(&status); equal {
		return nil
	}
```